core/src/main/java/org/elasticsearch/rest/action/support/RestStatusToXContentListener.java

```diff
@@ -23,14 +23,30 @@
 import org.elasticsearch.rest.BytesRestResponse;
 import org.elasticsearch.rest.RestChannel;
 import org.elasticsearch.rest.RestResponse;
+import org.elasticsearch.rest.RestStatus;
+
+import java.util.function.Function;
 
 /**
- *
+ * Content listener that extracts that {@link RestStatus} from the response.
  */
 public class RestStatusToXContentListener<Response extends StatusToXContent> extends RestResponseListener<Response> {
+    private final Function<Response, String> extractLocation;
```

Comment:
I went this route because I'd like to do a followup that adds the `Location` header to other places and this seemed a super convenient place to do it.